flowr-compilers/src/main/java/com/fueled/flowr/compilers/DeepLinkAnnotationCompiler.java

```diff
@@ -0,0 +1,306 @@
+    private static final String FLOWR_INTERNAL_PACKAGE_NAME = "com.fueled.flowr.internal";
+
+    private static final TypeVariableName fragmentTypeVariableName = TypeVariableName.get("T", ClassName.get("android.support.v4.app", "Fragment"), ClassName.get(FLOWR_PACKAGE_NAME, "FlowrFragment"));
+    private static final ParameterizedTypeName classExtendsFragmentTypeName = ParameterizedTypeName.get(ClassName.get(Class.class), WildcardTypeName.subtypeOf(fragmentTypeVariableName));
+    private static final ClassName FlowrDeepLinkInfoClassName = ClassName.get(FLOWR_INTERNAL_PACKAGE_NAME, "FlowrDeepLinkInfo");
+
+    private static final ClassName bundleClassName = ClassName.get("android.os", "Bundle");
+    private static final ClassName uriClassName = ClassName.get("android.net", "Uri");
+    private static final ParameterizedTypeName listStringTypeName = ParameterizedTypeName.get(List.class, String.class);
+
+    /**
+     * Check if a list of string contains the same substring.
+     *
+     * @param strings The list of String to search
+     * @param pos     The position until where the character are the same.
+     * @return
+     */
+    private static boolean allCharactersAreSame(List<String> strings, int pos) {
+        String first = strings.get(0);
+        for (String curString : strings) {
+            if (curString.length() <= pos
+                    || curString.charAt(pos) != first.charAt(pos)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Create the JavaPoet Type builder for the FlowrDeepLinkHandlerImpl class.
+     *
+     * @return The builder for the FlowrDeepLinkHandlerImpl class.
+     */
+    private static TypeSpec.Builder getClassObject() {
+        return TypeSpec.classBuilder(HANDLER_FILE_NAME)
+                .addTypeVariable(fragmentTypeVariableName)
+                .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
+                .addSuperinterface(ClassName.get(FLOWR_INTERNAL_PACKAGE_NAME, "FlowrDeepLinkHandler"));
+    }
+
+    /**
+     * Create the bundleUriInfo method of FlowrDeepLinkHandlerImpl class. This method extract the variables from a URL and add them to a Bundle.
+     * The structure is <variable, value>.
+     * For example the pattern in the DeepLink annotation \"/user/{userId}\", if the user access the app through \"/user/1234\", the function will return <userId,1234>.
+     * Please note that all the variable will a string.
+     *
+     * @return a Bundle that contains the variables from the deep link.
+     */
+    private static MethodSpec bundleUriInfo() {
+        return MethodSpec.methodBuilder("bundleUriInfo")
+                .addModifiers(Modifier.PRIVATE, Modifier.STATIC)
+                .returns(bundleClassName)
+                .addParameter(uriClassName, "uri")
+                .addParameter(String.class, "pattern")
+                .addCode("        String regex = getRegexPattern(pattern);\n" +
```

Comment:
@chetan-fueled no we can't, because the annotation compiler is executed in the `sample` package and not in the `compiler` package, so you cannot access resources from the `compiler` package. That's the reason for the creation and use of JavaPoet